topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -554,7 +568,9 @@ private void accumulateResults(Map<ReconcilableTopic, Either<TopicOperatorExcept
         });
         actuallyDifferentRf.ok().forEach(pair -> {
             var reconcilableTopic = pair.getKey();
-            var partitions = pair.getValue().partitionsWithDifferentRfThan(reconcilableTopic.kt().getSpec().getReplicas());
+            var specPartitions = reconcilableTopic.kt().getSpec().getPartitions() != null
+                ? reconcilableTopic.kt().getSpec().getPartitions() : BROKER_DEFAULT;
```

Comment:
This duplicate the expression on line 302. Can we factor out a method for it?

Reply:
Sure.